### PR TITLE
Provisional PR with monorepo-wide fixes

### PR DESCRIPTION
**Pull request by Oaphi**
This is a provisional pull request that fixes some of the compilation issues due to TypeScript updates, including:

- `paths` and `baseUrl` fields to relevant tsconfig.json files to ensure the compiler understands `@chatbot/*` imports
- several explicit imports from the `url` (native) and `node-fetch` (package) modules
- one `@ts-ignore`  for `bot.permissionCheck` in an `if` statement as the former returns `Promise<boolean>` (needs further investigation to ensure that `await`ing it does not break existing functionality that ended up relying on the bug
- removal of `abstract async` as `async` is not allowed in `abstract` members
- organized imports (automatic action)
- type guards for the `catch` block as TypeScript made the parameter `unknown` by default in `strict` mode recently, which was a breaking change
- added `lib` fields where necessary in tsconfig.json files to ensure the compiler allows new-ish features (such as `matchAll` on strings)

### CODE REVIEW

File: packages/plugins/src/plugins/eval.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const evalPlugin: PluginFunction = (bot) => {
             msg.info.rawContent.replace(/<br>/g, '\n').replace(/<.+>/g, '')
         );
         if (
+            // @ts-ignore FIXME: can RegisterHandler accept async callbacks?
```
Comment (jbis9051) [inline packages/plugins/src/plugins/eval.ts:51]: Yes, it should be able to

File: packages/plugins/src/plugins/eval/index.ts [hunk 2]
```diff
@@ -76,13 +76,15 @@ export default function (
...
                 .runSync(context, { timeout });
         } catch (e) {
-            sendData(
-                false,
-                JSON.stringify(e.toString()),
-                JSON.stringify([]),
-                start,
-                Date.now()
-            );
+            if (typeof e === "object" && e) {
```
Comment (jbis9051) [inline packages/plugins/src/plugins/eval/index.ts:79]: What if it's not an object?

Comment (Oaphi) [reply packages/plugins/src/plugins/eval/index.ts]: As discussed, switching to `Object.prototype.toString.call(e)` in 4cdf7b731b8f364377d24c974fd755e9a627a1e8

File: packages/plugins/src/plugins/eval.ts (hunk 1 above)
Comment (Oaphi) [reply packages/plugins/src/plugins/eval.ts]: Thanks for reassuring! Resolved in 975cba6cb2807af0c4c5aab4d96cc388fc51db50

**Pull request merged**